Documentation/Internals/MakefileSupport.md

```diff
@@ -0,0 +1,29 @@
+# Makefile Generation Support
+
+SwiftPM has support for generating makefiles for Simple Swift packages.  Some of
+the advanced features like modulemap generation, package dependencies, etc are
+out-of-scope unless required in future. The makefile support is accessible using
+an hidden flag and is intended to be used only for Swift.org related projects for
+bootstrapping them on the Swift CI.
+
+To generate makefile for a package, run the following command:
+
+```sh
+$ swift package __internal-tools --generate-makefile
```

Comment:
Just wondering: there is a `generate-xcodeproj` command for `swift package`, did you consider making this one similar, e.g. `generate-makefile`?

Reply:
Yeah, thats probably more correct/consistent.

Reply:
I'd probably go for make it easy to spell, but undocumented. That is a pattern the compilers already use. Could also give it a leading underscore, if we want to make it even more clear.

Reply:
Cool, I'll use `swift package _generate-makefile`.